lang/step.md

```diff
@@ -190,7 +190,7 @@ impl<M: Memory> Machine<M> {
             Type::Union { fields, .. } => fields[field].0,
             _ => panic!("field projection on non-projectable type"),
         };
-        assert!(offset < ty.size::<M>());
+        assert!(offset < ty.size::<M>() || ty.size::<M>().is_zero());
```

Comment:
I think this will still fail when accessing the second field of `(i8, ())`.
Maybe get rid of this assert altogether?